src/lib.rs

```diff
@@ -98,6 +98,33 @@ impl Client {
         self.drive_receive(parse_ascii_response).await
     }
 
+    pub(crate) async fn get_write_multi_responses(
+        &mut self,
+        num_results: usize,
+    ) -> Result<Response, Error> {
+        let mut results: Vec<Result<(), Error>> = Vec::new();
+
+        for _ in 0..num_results {
```

Comment:
Can all of this method be turned into a single iterator? Otherwise we are duplicating a lot of work.